File: app/src/main/java/dagger/SchedulerModule.java
package dagger;

import javax.inject.Singleton;
import org.kinecosystem.kinit.util.AndroidScheduler;
import org.kinecosystem.kinit.util.Scheduler;

@Module
public class SchedulerModule {

    @Provides
    @Singleton
    public Scheduler scheduler() {
        return new AndroidScheduler();
    }
}

File: app/src/main/java/org/kinecosystem/tippic/dagger/SchedulerModule.kt
package org.kinecosystem.tippic.dagger

import dagger.Module
import dagger.Provides
import org.kinecosystem.tippic.util.AndroidScheduler
import org.kinecosystem.tippic.util.Scheduler
import javax.inject.Singleton


@DebugOpenClass
@Module
class SchedulerModule {

    @Provides
    @Singleton
    fun scheduler(): Scheduler = AndroidScheduler()
}
